flink-table/flink-table-planner/src/main/scala/org/apache/flink/table/codegen/ExpressionReducer.scala

```diff
@@ -117,8 +128,18 @@ class ExpressionReducer(config: TableConfig)
       generatedFunction.code)
     val function = clazz.newInstance()
 
-    // execute
-    val reduced = function.map(EMPTY_ROW)
+    val richMapFunction = function match {
+      case r: RichMapFunction[Row, Row] => r
+      case _ => throw new TableException("RichMapFunction[Row, Row] required here")
+    }
+
+    val reduced = try {
+      richMapFunction.open(parameters)
+      // execute
+      richMapFunction.map(EMPTY_ROW)
+    } finally {
+      richMapFunction.close()
+    }
```

Comment:
This can be simplified to
```
  val reduced = try {
      FunctionUtils.openFunction(function, parameters)
      function.map(EMPTY_ROW)
    } finally {
      FunctionUtils.closeFunction(function)
    }
```

Reply:
Hi @aljoscha, Thank you for your advise. I have fixed it.